user/signer.go

```diff
@@ -48,3 +48,40 @@ func NewSignerRFC6979WithID(pk ecdsa.PrivateKey, id ID) *SignerRFC6979 {
 func (s SignerRFC6979) UserID() ID {
 	return s.userID
 }
+
+// StaticSigner emulates real sign and contains already precalculated hash.
+// Provides [Signer] interface.
+type StaticSigner struct {
+	neofscrypto.StaticSigner
+	id ID
+}
+
+// NewStaticSigner creates new StaticSigner.
+func NewStaticSigner(scheme neofscrypto.Scheme, sig []byte, pubKey neofscrypto.PublicKey) *StaticSigner {
+	var id ID
+
+	switch t := pubKey.(type) {
```

Comment:
`default: id = ID{}` - expected?

see also https://github.com/nspcc-dev/neofs-sdk-go/pull/479/commits/257a398926b635d38c8970a4e20c012e28d41877

Reply:
no, not expected. We may throw the `panic` if type isn't good for us, for instance _PublicKeyWalletConnect_

Reply:
In StaticSigner, unfortunately, we don't have the private key

Reply:
lets not shrink from abstraction to particular 

we can work with `ecdsa.PublicKey` and generic constructor like in mentioned commit

Reply:
Updated